pytest_bdd/feature.py

```diff
@@ -293,7 +293,11 @@ def __init__(self, basedir, filename, encoding="utf-8", strict_gherkin=True):
                     continue
                 mode = get_step_type(clean_line) or mode
 
-                if not scenario and prev_mode not in (types.BACKGROUND, types.GIVEN) and mode in types.STEP_TYPES:
+                allowed_prev_mode = (types.BACKGROUND, types.GIVEN)
+                if not strict_gherkin:
+                    allowed_prev_mode += (types.WHEN, )
```

Comment:
Could you explain what do you mean here?
"normal scenarios" - with disabled strict_gherkin or something else?
I can add test for it, but please describe the case.